lib/chef/cookbook/syntax_check.rb

```diff
@@ -248,8 +248,8 @@ def validate_ruby_file_inline(ruby_file)
       # Debugs ruby syntax errors by printing the path to the file and any
       # diagnostic info given in +error_message+
       def invalid_ruby_file(ruby_file, error_message)
-        file_relative_path = ruby_file[/^#{Regexp.escape(cookbook_path + File::Separator)}(.*)/, 1]
```

Comment:
This looks more like its a bug in this regular expression that is intended to strip the cookbook_path off the `ruby_file` and then return the remaining string.  That isn't the same as just taking the basename though since the whole path after the cookbook_path needs to be preserved.